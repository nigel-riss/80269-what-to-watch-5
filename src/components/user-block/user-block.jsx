import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {
  AppRoute,
  AuthorizationStatus,
} from '../../const.js';
import {redirectToRoute} from '../../store/actions/actions.js';


const UserBlock = (props) => {
  const {
    onAvatarClick,
    authorizationStatus,
  } = props;

  return (
    <div className="user-block">
      {authorizationStatus === AuthorizationStatus.AUTH
        ? <div
          className="user-block__avatar"
          onClick={onAvatarClick}
        >
          <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
        </div>
        : <Link to={AppRoute.LOGIN} className="user-block__link">Sign in</Link>
      }
    </div>
  );
};


UserBlock.propTypes = {
  onAvatarClick: PropTypes.func.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
};


const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
});

const mapDispatchToProps = (dispatch) => ({
  onAvatarClick() {
    dispatch(redirectToRoute(AppRoute.MY_LIST));
  }
});


export {UserBlock};
export default connect(mapStateToProps, mapDispatchToProps)(UserBlock);
