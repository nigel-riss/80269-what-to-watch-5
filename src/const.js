const FilmTab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

const RatingLimits = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  VERY_GOOD: 10,
};

const HOVER_PLAY_DELAY = 1000;
const FILM_CARD_HEIGHT = 175;
const FILM_CARD_WIDTH = 280;
const FILMS_SHOWN_PER_CLICK = 8;

export {
  FILM_CARD_HEIGHT,
  FILM_CARD_WIDTH,
  FILMS_SHOWN_PER_CLICK,
  FilmTab,
  HOVER_PLAY_DELAY,
  RatingLimits,
};
