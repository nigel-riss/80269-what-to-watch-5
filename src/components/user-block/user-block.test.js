import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {UserBlock} from './user-block.jsx';
import {AuthorizationStatus} from '../../const.js';


const noop = () => {};

it(`Should UserBlock render correctly`, () => {
  const tree = renderer
    .create(
        <MemoryRouter>
          <UserBlock
            authorizationStatus={AuthorizationStatus.AUTH}
            onAvatarClick={noop}
            userAvatarUrl={`/img/avatar.png`}
          />
        </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
