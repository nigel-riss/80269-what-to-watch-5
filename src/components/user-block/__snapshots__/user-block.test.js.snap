// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserBlock render correctly 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
    onClick={[Function]}
  >
    <img
      alt="User avatar"
      height="63"
      src="/img/avatar.png"
      width="63"
    />
  </div>
</div>
`;
